flink-tests/src/test/java/org/apache/flink/test/classloading/ClassLoaderITCase.java

```diff
@@ -282,10 +278,12 @@ public void testCheckpointingCustomKvStateJobWithCustomClassLoader() throws IOEx
 			Collections.singleton(new Path(CHECKPOINTING_CUSTOM_KV_STATE_JAR_PATH)),
 			Collections.emptyList());
 
-		expectedException.expectCause(
-			Matchers.hasProperty("cause", isA(SuccessException.class)));
-
-		program.invokeInteractiveModeForExecution();
+		try {
+			program.invokeInteractiveModeForExecution();
+			fail("exception should happen");
+		} catch (Exception e) {
```

Comment:
It's enough to catch `ProgramInvocationException` here. All other exceptions will fail the test anyways.

Strictly speaking, we relax the assertion a bit. `SuccessException` is now allowed to appear anywhere in the causal chain, whereas the test before was asserting:

```
assertThat(e.getCause().getCause(), instanceOf(SuccessException.class));
```

Reply:
Exception type is updated. 
This assertion relaxing should not be a problem, since this case to verify that task in the job can start and restore state successfully, otherwise the `SuccessException` will not be thrown.

Reply:
ok